src/docs/resources/architectural-overview.md

```diff
@@ -0,0 +1,966 @@
+custom platform embedder, as in [this stripped-down
+example](https://github.com/chinmaygarde/fluttercast) that supports remoting
+Flutter sessions through a VNC-style framebuffer or [this worked example for
+Raspberry Pi](https://github.com/ardera/flutter-pi).
+
+Each platform has its own set of APIs and constraints. Some brief
+platform-specific notes:
+
+- On iOS and macOS, Flutter is loaded into the embedder as a `UIViewController`
+  or `NSViewController`, respectively. The platform embedder creates a
+  `FlutterEngine`, which serves as a host to the Dart VM and your Flutter
+  runtime, and a `FlutterViewController`, which attaches to the `FlutterEngine`
+  to pass UIKit or Cocoa input events into Flutter and to display frames
+  rendered by the `FlutterEngine` using Metal or OpenGL.
+- On Android, Flutter is, by default, loaded into the embedder as an `Activity`.
+  The view is controlled by a
+  [`FlutterView`](https://api.flutter.dev/javadoc/io/flutter/embedding/android/FlutterView.html),
+  which renders Flutter content either as a view or a texture, depending on the
+  composition and z-ordering requirements of the Flutter content.
+- On Windows, Flutter is hosted in a traditional Win32 app, and content is
+  rendered using
+  [ANGLE](https://chromium.googlesource.com/angle/angle/+/master/README.md), a
+  library that translates OpenGL API calls to the DirectX 11 equivalents.
+  Efforts are currently underway to also offer a Windows embedder using the UWP
+  app model, as well as to replace ANGLE with a more direct path to the GPU via
+  DirectX 12.
+
+## Integrating with other code
+
+Flutter provides a variety of interoperability mechanisms, whether you’re
+accessing code or APIs written in a language like Kotlin or Swift, calling a
+native C-based API, embedding native controls in a Flutter app, or embedding
+Flutter in an existing application.
+
+### Platform channels
+
+For mobile and desktop apps, Flutter allows you to call into custom code through
+a _platform channel_, which is a simple mechanism for communicating between your
+Dart code and the platform-specific code of your host app. By creating a common
+channel (encapsulating a name and a codec), you can send and receive messages
+between Dart and a platform component written in a language like Kotlin or
+Swift. Data is serialized from a Dart type like `Map` into a standard format,
+and then deserialized into an equivalent representation in Kotlin (such as
+`HashMap`) or Swift (such as `Dictionary`).
+
+![How platform channels allow Flutter to communicate with host
+code](../../images/arch-overview/platform-channels.svg){:width="100%"}
+
+The following is a simple platform channel example of a Dart call to a receiving
+event handler in Kotlin (Android) or Swift (iOS):
+
+```dart
+// Dart side
+const channel = MethodChannel('foo');
+final String greeting = await channel.invokeMethod('bar', 'world');
+print(greeting);
+```
+
```

Comment:
You don't need to skip Swift, Kotlin, Java, or Objective-C. Travis doesn't evaluate these.